src/dash-table/derived/filter/map.ts

```diff
@@ -61,20 +61,49 @@ export default memoizeOneFactory((
     return newMap;
 });
 
-export const updateMap = (
-    map: Map<string, SingleColumnSyntaxTree>,
-    column: IVisibleColumn,
-    value: any
-): Map<string, SingleColumnSyntaxTree> => {
-    const safeColumnId = column.id.toString();
-
+function updateMap(map: Map<string, SingleColumnSyntaxTree>, column: IVisibleColumn, value: any) {
+    const id = column.id.toString();
     const newMap = new Map<string, SingleColumnSyntaxTree>(map);
 
     if (value && value.length) {
-        newMap.set(safeColumnId, new SingleColumnSyntaxTree(value, column));
+        newMap.set(id, new SingleColumnSyntaxTree(value, column));
     } else {
-        newMap.delete(safeColumnId);
+        newMap.delete(id);
     }
 
     return newMap;
+}
```

Comment:
Isolating the map update logic - can be called for a single column update (user edits filter) or through header `clear` and `delete` column actions (which may impact multiple columns at the same time)